File: backend/api/runtime.py
"""
Runtime execution API.
"""


from fastapi import APIRouter

from core.api import (
    ApiResponse
)
from core.container import (
    get_dependency,
    DependencyType
)


from core.runtime import (
    AgentTask
)
from core.runtime import (
    RuntimeHealthChecker
)


router = APIRouter(
    prefix="/runtime",
    tags=["Runtime"]
)



@router.post("/execute")
async def execute_agent(
    goal:str
):


    runtime_manager = get_dependency(

        DependencyType.RUNTIME

    )


    runtime = runtime_manager.get_runtime()



    task = AgentTask(

        task_id="task_001",

        goal=goal

    )


    result = await runtime.execute(
        task
    )


    return ApiResponse(

        data=result

    ).to_dict()


@router.get("/health")
async def runtime_health():


    runtime_manager = get_dependency(

        DependencyType.RUNTIME

    )


    runtime = (
        runtime_manager.get_runtime()
    )


    checker = RuntimeHealthChecker(
        runtime
    )


    return await checker.check()

@router.get("/planner")
async def planner_status():


    runtime_manager = get_dependency(

        DependencyType.RUNTIME

    )


    runtime = (
        runtime_manager.get_runtime()
    )


    return {

        "planner_enabled":

            runtime.planner is not None

    }

File: frontend/src/api/runtime.ts
import { request, type ApiEnvelope } from "@/api/client";
import { ENDPOINTS } from "@/api/endpoints";

/** Mirrors RuntimeHealthChecker.check() exactly (core/runtime/*health*.py). */
export interface RuntimeHealth {
  runtime: boolean;
  provider: boolean;
  memory: boolean;
  tools: boolean;
  healthy: boolean;
}

export interface PlannerStatus {
  planner_enabled: boolean;
}

export function getRuntimeHealth(signal?: AbortSignal): Promise<RuntimeHealth> {
  return request<RuntimeHealth>(ENDPOINTS.runtimeHealth, { signal });
}

export function getPlannerStatus(signal?: AbortSignal): Promise<PlannerStatus> {
  return request<PlannerStatus>(ENDPOINTS.runtimePlanner, { signal });
}

/**
 * Dispatches a real agent run. `goal` is a plain query param on the
 * backend (async def execute_agent(goal: str), no Pydantic body model),
 * not a JSON body — matched here exactly. The result shape is whatever
 * AgentTask execution returns, which core/runtime doesn't expose a
 * schema for, so it's typed as unknown rather than guessed.
 */
export function executeAgentTask(goal: string, signal?: AbortSignal): Promise<ApiEnvelope<unknown>> {
  return request<ApiEnvelope<unknown>>(ENDPOINTS.runtimeExecute, {
    method: "POST",
    query: { goal },
    signal,
  });
}
